### DISCUSSION TITLE
Allows Rules to be run with Board?

### DISCUSSION THREAD

**Discussion by mamamia5x** [BattlesnakeOfficial/feedback#7] [Feature Requests]
It would be cool to have the CLI to be run within board.
Could there be any implementation of this? Could someone have board and then create a game via the CLI and then have the board output the game?

**Comment by Nettogrof** [comment]
Currently isn't possible, but I agree I wish to be able to run a game locally  (example my snake v1 against my snake v2 to compare them)  and see the game with the board.

**Comment by andrewdmaclean** [comment]
Hey @mamamia5x 👋 Just wanted to reply to this to bump it up on the list in case anyone else might be interested in the enhancement.

**Comment by xtagon** [comment]
Previous engine versions were available that did this, they ran a local version of the engine and had a board viewer for testing. I really miss it but I understand why it was necessary to close parts of the engine.

The board viewer is available and is open source, but it does seem rather coupled to the engine websocket interface, so it may be possible to do this but it's not straightforward.

https://github.com/BattlesnakeOfficial/board

**Reply by bvanvugt** [reply]
Yeah, it really isn't to do in current state. I think a strategy here would be to more more pieces of the engine into a go module (or the rules repo) - ones that are responsible for the websocket api, and maybe even data manipulation from ruleset to Move request.

An alternative would be to beef up something like Mojave with official rules & game board support.

**Comment by eferri** [comment]
I've implemented a basic websocket server that can be used to view a game with a local instance of the board viewer. The code is in a fork of the CLI here: https://github.com/eferri/rules . 

Would there be interest in merging this in? I've only tested the `solo` ruleset so far, but it shouldn't be hard to confirm it works with the other game modes.
